skbeam/core/roi.py

```diff
@@ -423,15 +423,30 @@ def circular_average(image, calibrated_center, threshold=0, nx=100,
         Left edge of first bin defaults to minimum value of x
     max_x : float, optional number of pixels
         Right edge of last bin defaults to maximum value of x
+    mask : mask for 2D data. Assumes 1 is non masked and 0 masked.
+        None defaults to no mask.
+    
     Returns
     -------
     bin_centers : array
         The center of each bin in R. shape is (nx, )
     ring_averages : array
         Radial average of the image. shape is (nx, ).
+
+    See Also
+    --------
+    bad_to_nan_gen : Create a mask with np.nan entries
+
+    Notes
+    -----
     """
     radial_val = utils.radial_grid(calibrated_center, image.shape, pixel_size)
 
+    if mask is not None:
```

Comment:
I would go with:
```python
if not mask:
    mask = np.ones(image.shape)
image = image[mask]
radial_val = radial_val[mask]
```
This should take care of everything, and then you don't need the `np.ravel`s in the code below.